ios/ios_pad/source/controllers/switch_controller.c

```diff
@@ -465,11 +465,11 @@ static void handle_command_response(Controller* controller, SwitchCommandRespons
 
         // set the leds now that we know the device type
         setPlayerLeds(controller);
-
+    } else if (resp->command == SWITCH_COMMAND_SET_PLAYER_LEDS) {
```

Comment:
Note that `setPlayerLeds` is also called by `controllerSetLed_switch`, which would now redo the init sequence. There should probably be an early return in `handle_command_response` based on `controller->isReady`.